test/bench/synthetic_benchmarks.cpp

```diff
@@ -0,0 +1,277 @@
+{
+    const auto counter =
+        push("ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff01");  // -255
+    const auto jumpdest_offset = counter.size();
+    return counter + OP_JUMPDEST + inner_code +  // loop label + inner code
+           push(1) + OP_ADD + OP_DUP1 +          // counter += 1
+           push(jumpdest_offset) + OP_JUMPI;     // jump to jumpdest_offset if counter != 0
+}
+
+bytes_view generate_code(CodeParams params)
+{
+    static std::map<CodeParams, bytecode> cache;
+
+    auto& code = cache[params];
+    if (!code.empty())
+        return code;
+
+    code = generate_loop_v2(generate_loop_inner_code(params));  // Cache it.
+    return code;
+}
+}  // namespace
+
+void register_synthetic_benchmarks()
+{
+    std::vector<CodeParams> params_list;
+
+    // Nops & unops.
+    for (const auto opcode : {OP_JUMPDEST, OP_ISZERO, OP_NOT})
+        params_list.push_back({opcode, Mode::min_stack});
+
+    // Binops.
+    for (const auto opcode : {OP_ADD, OP_MUL, OP_SUB, OP_SIGNEXTEND, OP_LT, OP_GT, OP_SLT, OP_SGT,
+             OP_EQ, OP_AND, OP_OR, OP_XOR, OP_BYTE, OP_SHL, OP_SHR, OP_SAR})
+        params_list.insert(
+            params_list.end(), {{opcode, Mode::min_stack}, {opcode, Mode::full_stack}});
+
+    // Nullops.
+    for (const auto opcode : {OP_ADDRESS, OP_CALLER, OP_CALLVALUE, OP_CALLDATASIZE, OP_CODESIZE,
+             OP_RETURNDATASIZE, OP_PC, OP_MSIZE, OP_GAS})
+        params_list.insert(
+            params_list.end(), {{opcode, Mode::min_stack}, {opcode, Mode::full_stack}});
+
+    // PUSH.
+    for (auto opcode = OP_PUSH1; opcode <= OP_PUSH32; opcode = static_cast<evmc_opcode>(opcode + 1))
+        params_list.insert(
+            params_list.end(), {{opcode, Mode::min_stack}, {opcode, Mode::full_stack}});
+
+    // SWAP.
+    for (auto opcode = OP_SWAP1; opcode <= OP_SWAP16; opcode = static_cast<evmc_opcode>(opcode + 1))
+        params_list.insert(params_list.end(), {{opcode, Mode::min_stack}});
+
+    // DUP.
+    for (auto opcode = OP_DUP1; opcode <= OP_DUP16; opcode = static_cast<evmc_opcode>(opcode + 1))
+        params_list.insert(
+            params_list.end(), {{opcode, Mode::min_stack}, {opcode, Mode::full_stack}});
+
+
+    for (auto& [vm_name, vm] : registered_vms)
+    {
+        RegisterBenchmark((std::string{vm_name} + "/execute/synth/loop_v1").c_str(),
+            [&vm = vm](State& state) { execute(state, vm, generate_loop_v1({})); });
+        RegisterBenchmark((std::string{vm_name} + "/execute/synth/loop_v2").c_str(),
+            [&vm = vm](State& state) { execute(state, vm, generate_loop_v2({})); });
+    }
+
+    for (const auto params : params_list)
+    {
+        for (auto& [vm_name, vm] : registered_vms)
+        {
+            RegisterBenchmark(
+                (std::string{vm_name} + "/execute/synth/" + to_string(params)).c_str(),
+                [&vm = vm, params](State& state) { execute(state, vm, generate_code(params)); })
```

Comment:
This should be possible?
```suggestion
                [&vm = vm, code = generate_code(params)](State& state) { execute(state, vm, code); })
```

Reply:
This way we will generate all bytecodes on process startup. Not sure this is good thing, but I will measure this.

Reply:
Why do you think including code generation into the benchmark of executing code can be a good thing?

Reply:
This code generation is outside of the benchmark loop so its time is not measured.

With the change the `evmone-bench` startup time in Debug increases from `0.01s` to `0.34s`, so I don't think this is good change.

Reply:
The benchmark loop is here: https://github.com/ethereum/evmone/blob/master/test/bench/helpers.hpp#L70-L75.
Everything outside does not contribute to the timings.